core/src/main/java/com/rudderstack/android/sdk/core/persistence/DefaultPersistenceProviderFactory.java

```diff
@@ -47,10 +48,11 @@ public PersistenceProvider create(Application application) {
             dbVersion = 1;
         }
         if (isEncrypted) {
-            if (encryptionKey == null) {
-                RudderLogger.logWarn("DBPersistentManager: isEncrypted is true but encryptionKey is null. Proceeding with null key");
+            if (TextUtils.isEmpty(encryptionKey)) {
+                RudderLogger.logWarn("DBPersistentManager: isEncrypted is true but encryptionKey is null or empty. Proceeding with unencrypted database");
+                isEncrypted = false;
             }
-            if (encryptedDbName == null) {
+            else if (encryptedDbName == null) {
```

Comment:
```suggestion
            if (encryptedDbName == null) {
```
Reason: It doesn't matter `encryptionKey` is empty or not, `encryptedDbName` can still be `null`, right? So we should handle that irrespective of `encryptionKey`.

Reply:
If isEncrypted is false, we will be creating unencrypted database, so anyhow encrypted database name will not be required.

Reply:
Just try simulating the above scenario (where encryptionKey and encryptedDbName are null), and you'll get an exception, over [here](https://github.com/rudderlabs/rudder-sdk-android/blob/11e3ac22e3e9d22dd9ac65b443330914df4cf584/core/src/main/java/com/rudderstack/android/sdk/core/persistence/DefaultPersistenceProvider.java#L94).

Please consider enhancing the case when DB name is `null`. We should assigning default values in that case.

Reply:
We should be having null checks